app/src/main/java/io/stanwood/mhwdb/feature/armors/vm/ArmorDetailsViewModel.kt

```diff
@@ -15,24 +12,16 @@ class ArmorDetailsViewModel @Inject constructor(private val dataProvider: ArmorD
         dataProvider.armorId = armorId
     }
 
-    val header: Observable<Armor> =
+    val data =
         dataProvider.data
-            .filter {
-                it is Resource.Success
-            }
-            .map {
-                (it as Resource.Success).data
-            }
-            .observeOn(AndroidSchedulers.mainThread())
+            .filter { it.data != null }
+            .map { it.data!! }
+            .observeOn(AndroidSchedulers.mainThread())!!
 
-    val items: Observable<List<ArmorDetailsItemViewModel>> =
-        dataProvider.data.map {
-            when (it) {
-                is Resource.Success -> listOf(ArmorDetailsItemViewModel(it.data.name))
-                else -> emptyList()
-            }
-        }
-            .observeOn(AndroidSchedulers.mainThread())
+    val items =
+        data.map {
+            listOf(ArmorDetailsItemViewModel(it.name))
+        }!!
```

Comment:
Those !! everywhere are kinda ugly.

Reply:
but valid

Reply:
Maybe we'll find a better solution in the future, the more I think about it the more I like just writing out the return type here... 🤔